Packs/EWS/Integrations/EwsExtensionEXOPowershellV2/EwsExtensionEXOPowershellV2.ps1

```diff
@@ -398,6 +576,125 @@ function GetEXOCASMailboxCommand
+        Write-Output $human_readable, $entry_context, $raw_response
+    }
+    else
+    {
+        Write-Output $human_readable, $null, $raw_response
+    }
+
+}
+
+function EXOGetTenantAllowBlockListCommand
+{
+    [CmdletBinding()]
+    Param (
+        [Parameter(Mandatory)][ExchangeOnlinePowershellV2Client]$client,
+        [hashtable]$kwargs
+    )
+    $entry = $kwargs.entry
+    $list_type = $kwargs.list_type
+    $list_subtype = $kwargs.list_subtype
+    $action = $kwargs.action
+    $no_expiration = if ($kwargs.no_expiration -eq "true") { $true } else { $false }
+    $expiration_date = $kwargs.expiration_date
+    $raw_response = $client.EXOGetTenantAllowBlockList(
+        $entry, $list_type, $list_subtype, $action, $no_expiration, $expiration_date
+    )
+    $human_readable = TableToMarkdown $raw_response "Results of $command"
+    if ($kwargs.set_context -eq "yes")
+    {
+        $entry_context = @{ "$script:INTEGRATION_ENTRY_CONTEXT.CurrentTenantBlocks" = $raw_response }
+        Write-Output $human_readable, $entry_context, $raw_response
+    }
+    else
+    {
+        Write-Output $human_readable, $null, $raw_response
+    }
+
+}
+
+function EXOCountTenantAllowBlockListCommand
+{
+    [CmdletBinding()]
+    Param (
+        [Parameter(Mandatory)][ExchangeOnlinePowershellV2Client]$client,
+        [hashtable]$kwargs
+    )
+    $list_type = $kwargs.list_type
+    $list_subtype = $kwargs.list_subtype
+    $m = $client.EXOGetTenantAllowBlockList($null, $list_type, $list_subtype, $null, $null, $null) | Measure-Object
+    $raw_response = [PSCustomObject]@{
+        ListType = $list_type
+        ListSubType = $list_subtype
+        Count = $m.Count
+    }
+    $human_readable = TableToMarkdown $raw_response "Results of $command"
+    if ($kwargs.set_context -eq "yes")
+    {
+        $entry_context = @{ "$script:INTEGRATION_ENTRY_CONTEXT.CurrentListCount" = $raw_response }
+        Write-Output $human_readable, $entry_context, $raw_response
+    }
+    else
+    {
+        Write-Output $human_readable, $null, $raw_response
+    }
+
+}
+
+function EXORemoveTenantAllowBlockListCommand
+{
+    [CmdletBinding()]
+    Param (
+        [Parameter(Mandatory)][ExchangeOnlinePowershellV2Client]$client,
+        [hashtable]$kwargs
+    )
+    $entries = $kwargs.entries
+    $ids = $kwargs.ids
+    $list_type = $kwargs.list_type
+    $list_subtype = $kwargs.list_subtype
+    if ($entries -and $ids)
+    {
+        ReturnError "Error: Please provide either entries by value OR IDs, not both."
+    }
+    $raw_response = $client.EXORemoveTenantAllowBlockList(
+        $entries, $ids, $list_type, $list_subtype
+    )
+    $human_readable = TableToMarkdown $raw_response "Results of $command"
+    if ($kwargs.set_context -eq "yes")
+    {
+        $entry_context = @{ "$script:INTEGRATION_ENTRY_CONTEXT.RemovedTenantBlocks" = $raw_response }
+        Write-Output $human_readable, $entry_context, $raw_response
+    }
```

Comment:
No need to write to context on "delete" commands

Reply:
This is intended to allow storing a list of deleted items for later confirmation in playbooks/scripts that deletion was successful.  I would suggest keeping this here unless it cannot be included.